mudslide/electronics.py

```diff
@@ -44,16 +47,21 @@ def compute(self, X: ArrayLike, couplings: Any = None, gradients: Any = None, re
         """
         raise NotImplementedError("ElectronicModels need a compute function")
 
-    def update(self, X: ArrayLike, couplings: Any = None, gradients: Any = None) -> 'ElectronicModel_':
+    def update(self, X: ArrayLike, electronics: Any = None, couplings: Any = None, gradients: Any = None) -> 'ElectronicModel_':
         """
         Convenience function that copies the present object, updates the position,
         calls compute, and then returns the new object
         """
         out = cp.copy(self)
+        if electronics:
+            self.reference = electronics.reference
```

Comment:
I think this will fail if not given an electronics and is also polluting the data in the model by permanently grabbing hold of the reference. Instead do
```suggestion
        ref = electronics.reference if electronics is not None else None
```
and then pass `ref` to the compute function.